refactor(ast): remove `.iter()` where not needed

camchenry opened the pull request:
this is just an experiment, I noticed while profiling these code paths came up a number of times:
<img width="1077" alt="Screenshot 2024-09-19 at 8 32 51 PM" src="https://github.com/user-attachments/assets/d0f4d703-c456-4344-8f57-81f0b3f491a0">

camchenry commented:
just trying some stuff, going to close this

Boshen commented:
This seems like a acceptable improvement. Any improvement that reduces something is 👍 from me.

camchenry commented:
@Boshen I didn't think about it like that, but makes sense to me. If you're amenable to it then, I'll re-open

camchenry committed:
refactor(ast): remove `.iter()` where not needed (#5904)

this is just an experiment, I noticed while profiling these code paths came up a number of times:
<img width="1077" alt="Screenshot 2024-09-19 at 8 32 51 PM" src="https://github.com/user-attachments/assets/d0f4d703-c456-4344-8f57-81f0b3f491a0">
